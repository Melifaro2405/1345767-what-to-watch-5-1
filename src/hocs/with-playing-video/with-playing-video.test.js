import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withPlayingVideo from "./with-playing-video";

export const film = {
  id: 1,
  preview: {
    src: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `The Grand Budapest Hotel`
  },
  moreInfo: {
    backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Comedy`,
    releaseDate: 2014,
    playPreviewSrc: `https://some-link`,
    playVideoSrc: `https://some-link`,
    isAddToMyList: false
  },
  overview: {
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
      becomes Gustave's friend and protege.`,
    rating: 8.9,
    ratingCount: 240,
    director: `Wes Andreson`,
    actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
  },
  details: {
    runtime: 99
  }
};

const MockComponent = (props) => {
  const {children} = props;
  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withPlayingVideo(MockComponent);

it(`withPlayingVideo is rendered correctly`, () => {
  const noop = () => {};
  const noopPromise = () => {
    return {
      catch: noop
    };
  };

  const tree = renderer.create(
      <MockComponentWrapped
        film={film}
        onExitButtonClick={noop}
      >
        <React.Fragment />
      </MockComponentWrapped>, {
        createNodeMock: () => {
          return {
            play: noopPromise,
            pause: noop,
            src: `test`,
          };
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
