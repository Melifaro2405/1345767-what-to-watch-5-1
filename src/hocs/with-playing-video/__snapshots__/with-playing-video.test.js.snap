// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPlayingVideo is rendered correctly 1`] = `
<div>
  <video
    className="player__video"
    onTimeUpdate={[Function]}
    poster="img/player-poster.jpg"
    src="https://some-link"
  />
</div>
`;
